custom/ewsghana/forms.py

```diff
@@ -109,7 +109,7 @@ def __init__(self, *args, **kwargs):
         self.fields['facility'].widget = FacilitiesSelectWidget(domain=domain, id='facility')
 
     def save(self, user, domain):
-        ews_extension = EWSExtension.objects.get_or_create(user_id=user.get_id)[0]
+        ews_extension = EWSExtension.objects.get_or_create(user_id=user.get_id, domain=domain)[0]
```

Comment:
ok, i see, i didn't realize these were web users.  i think it's still worth it to wrap all of this save code in a `CriticalSection` to avoid race conditions if someone clicks the save button twice or something.  so you can do something like:

    with CriticalSection(['ews-get-or-create-extension-%s-%s' % (user.get_id, domain)]):
        ...do save